tests/Helpers.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace GuzzleHttp\Tests;
+
+class Helpers
+{
+    public static function readObjectAttribute(object $object, string $attributeName)
+    {
+        $reflector = new \ReflectionObject($object);
+
+        do {
+            try {
+                $attribute = $reflector->getProperty($attributeName);
+
+                if (!$attribute || $attribute->isPublic()) {
+                    return $object->$attributeName;
+                }
+
+                $attribute->setAccessible(true);
+
+                try {
+                    return $attribute->getValue($object);
+                } finally {
+                    $attribute->setAccessible(false);
```

Comment:
I dont see why this is needed.

Reply:
Just copied from the original implementation. Could remove it I suppose.

Reply:
Okey. 
I thought it did not matter.
I'll see if I can find the reason why it was added.

Reply:
I think it's probably not needed too. Doesn't really matter though. Only test code, and it doesn't hurt.